plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/utils/AndroidUtils.java

```diff
@@ -263,6 +278,29 @@ private static GradleSourceSet convertVariantToGradleSourceSet(Project project,
       Object testVariant = invokeMethod(variant, "getTestVariant");
       gradleSourceSet.setHasTests(unitTestVariant != null || testVariant != null);
 
+      // extensions
+      Map<String, LanguageExtension> extensions = new HashMap<>();
+      boolean isJavaSupported = Arrays.stream(SourceSetUtils.getSupportedLanguages())
+          .anyMatch(l -> Objects.equals(l, SupportedLanguages.JAVA.getBspName()));
+      if (isJavaSupported) {
+        DefaultJavaExtension extension = new DefaultJavaExtension();
+
+        extension.setJavaHome(DefaultInstalledJdk.current().getJavaHome());
+        extension.setJavaVersion(DefaultInstalledJdk.current().getJavaVersion().getMajorVersion());
+
+        extension.setSourceDirs(sourceDirs);
+        extension.setGeneratedSourceDirs(generatedSources);
+        extension.setClassesDir(sourceOutputs.iterator().next());
+        extension.setCompileTaskName(compileTaskName);
+
+        extension.setCompilerArgs(compilerArgs);
+        extension.setSourceCompatibility(getSourceCompatibility(compilerArgs));
```

Comment:
I think that what we really need is the source compatibility and target compatibility.

Things like the java home and java version may not required since the android.jar already contains the types from jdk

Reply:
Addressed in https://github.com/microsoft/build-server-for-gradle/pull/173/commits/f129e626874179194621bca70bd645af966d2fb2

Reply:
The vscode-gradle needs to be updated accordingly.

1. https://github.com/microsoft/vscode-gradle/blob/07878470c8dcc8a61b153671c78357d04743fa10/extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerBuildSupport.java#L587-L590

Here, instead of throw an exception, we should allow the empty java home and gradle version. Just leave log that they are empty.

2. Here https://github.com/microsoft/vscode-gradle/blob/07878470c8dcc8a61b153671c78357d04743fa10/extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerBuildSupport.java#L516-L540. When the Java Home is empty, the code here is unnecessary

Reply:
Ah I see.

Question though: If the sources, and generated sources are optional since we are already passing it in the build target info, why do we add it to extensions in the first place? Shouldn't it be the same for other project type support (java and scala)?

Reply:
Because each language can have its own source/generated sources, that's the part of language feature in Gradle: https://github.com/microsoft/build-server-for-gradle/blob/395b97757da8d0d82bc2c68c4aca4331bea4e584/plugin/src/main/java/com/microsoft/java/bs/gradle/plugin/SourceSetsModelBuilder.java#L100-L102

Reply:
Updated vscode-gradle with https://github.com/microsoft/vscode-gradle/pull/1594/commits/9021f77659658c560a8da5905e1d65e7719e964b